controllers/component_build_controller.go

```diff
@@ -352,6 +353,32 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		buildStatus.Message = "done"
 		writeBuildStatus(&component, buildStatus)
 
+	case BuildRequestTriggerPaCBuildAnnotationValue:
+		buildStatus := readBuildStatus(&component)
+		if !(buildStatus.PaC != nil && buildStatus.PaC.State == "enabled") {
+			log.Info("Can't rerun push pipeline because Pipelines as Code isn't provisioned for the Component")
+			return ctrl.Result{}, nil
+		}
+
+		reconcileRequired, err := r.TriggerPaCBuild(ctx, &component)
+
+		if err != nil {
+			if boErr, ok := err.(*boerrors.BuildOpError); ok && boErr.IsPersistent() {
+				log.Error(err, "Failed to rerun push pipeline for the Component")
+				buildStatus := readBuildStatus((&component))
+				buildStatus.PaC.ErrId = boErr.GetErrorId()
+				buildStatus.PaC.ErrMessage = boErr.ShortError()
+				writeBuildStatus(&component, buildStatus)
+			} else {
+				// transient error, retry
+				log.Error(err, "Failed to rerun push pipeline for the Component with transient error")
+				return ctrl.Result{}, err
+			}
+		}
+		if reconcileRequired {
+			return ctrl.Result{Requeue: true}, nil
```

Comment:
It shouldn't reconcile after a permanent error.

Reply:
ok, reconciling now only when no error occured